pkg/apis/velero/v1/snapshot.go

```diff
@@ -77,4 +80,37 @@ const (
 
 	// SnapshotPhaseFailed means the volume snapshot was unable to execute.
 	SnapshotPhaseFailed SnapshotPhase = "Failed"
+
+	// SnapshotPhaseInProgress means the volume snapshot created, but isn't ready yet.
+	SnapshotPhaseInProgress SnapshotPhase = "InProgress"
 )
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Snapshot stores information about a persistent volume snapshot taken as
+// part of a Velero backup.
+type Snapshot struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// +optional
```

Comment:
I don't think the entire spec can be optional - some minimum would need to be defined in order to know what needs to be snapshotted or restored.